### ISSUE TITLE
When use full-sentence to make the pretrain data，if document1’s tokens num didn’t get to 512, and if add another sentence from document2, tokens num will exceed 512, how to deal with this circumstance?  

### ISSUE THREAD

**Issue by YuxiangLu** [pytorch/fairseq#1170]

**Comment by lematt1991** [comment]
I'm sorry, I don't understand the issue.  Can you provide more details as well as the command you are using.

**Comment by YuxiangLu** [comment]
> I'm sorry, I don't understand the issue. Can you provide more details as well as the command you are using.

That is described in the paper:
FULL - SENTENCES: Each input is packed with full sentences sampled contiguously from one or more documents, such that the total length is at most 512 tokens. Inputs may cross document boundaries. When we reach the end of one document, we begin sampling sentences from the next document and add an extra separator token between documents. We remove the NSP loss.
--------
I mean, when inputs are cross document boundaries,  and reach the end of one document, sampled sentences from the next  document, are this sampled sentences must  be full sentence, or can be fragment of a sentence?
